File: medis-service/src/main/java/io/mersys/medis/documents/patient/Allergy.java
package io.mersys.medis.documents.patient;

import javax.validation.constraints.NotBlank;

import org.springframework.data.annotation.TypeAlias;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@TypeAlias("Allergy")
@Data
@AllArgsConstructor
@NoArgsConstructor

public class Allergy{
	
	@NotBlank
	public String category;

	@NotBlank
	public String allergen;
	
	//@NotBlank
	//public String reaction;
	
	@NotBlank
	public String degree;
	
}	

File: medis-ui/src/app/main/patients/classes/allergy.ts
export class Allergy{
  constructor(public category?: string, public allergen?: string, public degree?: string){
      this.category = category ? category : '';
      this.allergen = allergen ? allergen : '';
      this.degree = degree ? degree : '';
  }

}
export enum Degree{
    HIGH = 'HIGH',
    MEDIUM = 'MEDIUM',
    LAW = 'LAW'
}
